### PR TITLE
Add vali spear

### PR DESCRIPTION
**Pull request by Waselon**
## About The Pull Request

Added a spear that can be connented to Vali module. 32 onehanded damage, 2 tiles reach.

## Why It's Good For The Game

More viable melee options

## Changelog
:cl:
add: Added Vali spear
/:cl:

### CODE REVIEW

**Comment by Mrrpip** [comment]
This reeks of powercreep.

It's the damage of a machete with harvester abilities and spear reach for first come, first serve.

**Comment by Waselon** [comment]
> This reeks of powercreep.
> 
> It's the damage of a machete with harvester abilities and spear reach for first come, first serve.
> 
> PS: The spear isn't even two handed.

If you dont like it and have any balance suggestions, please say them.

**Comment by Mrrpip** [comment]
> > This reeks of powercreep.
> > It's the damage of a machete with harvester abilities and spear reach for first come, first serve.
> > PS: The spear isn't even two handed.
> 
> If you dont like it and have any balance suggestions, please say them.

I thought it'd be obvious with you just taking the best of all 3 melee weapons and putting it in for free, but here's a few.

1: Don't make the damage as good as the machete, we don't want a period again where it's completely invalidated by the harvester variant being a copy but better. (It's alleviated by being relegated to 3 only, but that's unfun due to it being first-come, first-serve with no ability to get more from req).
2: If you are truly insistent on adding it at this power level, make it a req buy.
3: Give it two-handed functionality.
4: Make it do thrown damage....Like a spear.

Most limited weapons are both req-buys and sidegrades/gimmicks.

A.E, the T27 has it's whole plant it or it's going to spread everywhere, the tesla gun is the tesla gun.

This is not a side-grade, it's an up-grade. But is unique in that you can get it roundstart.

**Comment by Pill-U** [comment]
Machete has a force of 75 and harvester 60. I propose that spear should have about the same damage decrase in exchange for harvester bonuses.
Regular spear has a force of 40 and a force_wielded of 75 so harvester spear should have a force of around 25 and force_wielded of around 60.
You should also copy and paste throwforce, throw_speed, sharp, w_class, hitsound and attack_verb from regular spear if the harvester doesnt have those set which I believe it does not.

**Comment by Waselon** [comment]
> Machete has a force of 75 and harvester 60. I propose that spear should have about the same damage decrase in exchange for harvester bonuses. Regular spear has a force of 40 and a force_wielded of 75 so harvester spear should have a force of around 25 and force_wielded of around 60. You should also copy and paste throwforce, throw_speed, sharp, w_class, hitsound and attack_verb from regular spear if the harvester doesnt have those set which I believe it does not.

Unfortunately i dont have any experience in byond coding so I am unable to make it two-handed without copying hundreds lines of code from weapons/twohanded

**Comment by Pill-U** [comment]
Also you can't wield this spear since it's not a child of twohanded, which is part of the balance of the regular spear, as you either get the damage bonus or a free hand wield a shield with.
Only idea I got to fix this is to refactor entire melee weapon code (good luck lol) or making it a child of twohanded and copy and pasting the harvester code. Pretty ugly doe.

**Review by hyper2snyper — CHANGES_REQUESTED** [review]
Given the obvious balance concerns regarding reach and near equivalent force of the regular harvester I would say no, as is.
However if you made the force ~30-35 I would approve since you would be then trading damage for the reach.

**Comment by Pill-U** [comment]
I did some testing and doing it this way seems to work except for loading the spear with chemicals.
/obj/item/weapon/twohanded/spear/tactical/harvester
	name = "\improper HP-S Harvester spear"
	desc = "TerraGov Marine Corps' experimental High Point-Singularity 'Harvester' spear. An advanced weapon that has the ability to apply a variety of debilitating effects when loaded with certain reagents. Activate after loading to prime a single use of an effect. It also harvests substances from alien lifeforms it strikes when connected to the Vali system."
	flags_item = TWOHANDED | DRAINS_XENO

works: Spear can attach to vali, spear cannot be dropped while attached to vali, spear can extract blood from xenos into vali, spear can be wielded.
doesn't work: loading chemicals into spear.

**Comment by Pill-U** [comment]
mind posting the sprites for the spear btw

**Comment by Waselon** [comment]
> mind posting the sprites for the spear btw

**Comment by Pill-U** [comment]
Since I have no way to contact you otherwise, I'll just share the pastebin link here.
This is an absolutely shit way to implement this, but boy it works. If you are okay with doing it this way and maintainers are too, you just have to copy and paste and change what I recommended above. 
Tested on a private Server, didn't get any runtimes and it seems to work normally. You're welcome. Also I'd change the vending amount to -1 so it's infinite. Hardcapping the amount of something is not a good way to do balance imo.
https://pastebin.com/NpTjRQ0t

**Comment by Waselon** [comment]
> Since I have no way to contact you otherwise, I'll just share the pastebin link here. I basically just copy and pasted harvester code so this is an absolutely shit way to implement this, but boy it works. If you are okay with doing it this way and maintainers are too, you just have to copy and paste and change what I recommended above. Tested on a private Server, didn't get any runtimes and it seems to work normally. You're welcome. Also I'd change the vending amount to -1 so it's infinite. Hardcapping the amount of something is not a good way to do balance imo. https://pastebin.com/NpTjRQ0t

Thank you

**Comment by Waselon** [comment]
Made Vali spear twohanded;
Onehanded damage - 32, twohanded damage - 60 (exact 20% damage reduction like Harvester)

**Review by hyper2snyper — CHANGES_REQUESTED** [review]
I much prefer the other system. Dont do copy paste of the harvester code. I know this is the only way to get both but this isnt the right way to do it. Honestly these typepaths are a mess in general. No full on copy-paste

**Comment by Waselon** [comment]
> I much prefer the other system. Dont do copy paste of the harvester code. I know this is the only way to get both but this isnt the right way to do it. Honestly these typepaths are a mess in general. No full on copy-paste

If I remove copy-paste code, would you prefer a two-handed version that cant use reagents, or one-handed that can?

**Comment by Waselon** [comment]
I dont want to touch other's people work, but what if we move harvester sword code into weapons/twohanded.dm, so we dont copypaste anything

**Comment by Pill-U** [comment]
Realized this yesterday but didn't have time to tell you, sorry. My idea was overly complicated. Your original version prob works if you just do
flags_item = ISWIELDED | DRAINS_XENO
try making the spear /obj/item/weapon/claymore/harvester/spear again and just copying its attributes except for the harvester code.

**Comment by AndroidMK2** [comment]
I'm fine with this PR but the spear looks like a knife strapped to a black stick rather than something mass produced for combat

**Comment by Waselon** [comment]
> Realized this yesterday but didn't have time to tell you, sorry. My idea was overly complicated. Your original version prob works if you just do flags_item = ISWIELDED | DRAINS_XENO try making the spear /obj/item/weapon/claymore/harvester/spear again and just copying its attributes except for the harvester code.

ISWIELDED flag doesnt exist. There are WIELDED flag but it does nothing

**Comment by Pill-U** [comment]
sorry I meant TWOHANDED

**Comment by Waselon** [comment]
> sorry I meant TWOHANDED

nah it doesnt work either

**Comment by Waselon** [comment]
Alright, i've changed Harvester blade base type to weapons/twohanded. It is still onehanded and deals same amount af damage, so it shouldnt affect anything. Now Harvester spear can properly inherit blade and be twohanded.

File: code/game/objects/machinery/vending/marine_vending.dm [hunk 1]
```diff
@@ -103,6 +103,7 @@
 			/obj/item/stack/throwing_knife = -1,
 			/obj/item/weapon/twohanded/spear/tactical = -1,
 			/obj/item/ammo_magazine/standard_smartmachinegun = 4,
+			/obj/item/weapon/claymore/harvester/spear = -1,
```
Comment (hyper2snyper) [inline code/game/objects/machinery/vending/marine_vending.dm:106]: minor suggestion here, perhaps move it next to the harvester sword.

File: code/game/objects/items/weapons/blades.dm [hunk 2]
```diff
@@ -42,16 +42,23 @@
 	return(BRUTELOSS)
 
 /obj/item/weapon/claymore/harvester
+	parent_type = /obj/item/weapon/twohanded
```
Comment (hyper2snyper) [inline code/game/objects/items/weapons/blades.dm:45]: This wreaks of shitcode. Ima have to ask about this. Does it work?

File: code/game/objects/items/weapons/blades.dm [hunk 3]
```diff
@@ -196,6 +203,20 @@
...
 	return ..()
 
+/obj/item/weapon/claymore/harvester/spear
+	name = "\improper HP-S Harvester spear"
+	desc = "TerraGov Marine Corps' experimental High Point-Singularity 'Harvester' spear. An advanced weapon that trades sheer force for the ability to apply a variety of debilitating effects when loaded with certain reagents. Activate after loading to prime a single use of an effect. It also harvests substances from alien lifeforms it strikes when connected to the Vali system."
+	icon_state = "vali_spear"
+	item_state = "vali_spear"
+	force = 32
+	flags_equip_slot = ITEM_SLOT_BACK
+	force_wielded = 60
```
Comment (hyper2snyper) [inline code/game/objects/items/weapons/blades.dm:213]: Its parent already declares `force_wielded` as 60, so you wouldnt need it here.

**Review by hyper2snyper — CHANGES_REQUESTED** [review]
I will get back to you on the `parent_type` thing

File: code/game/objects/items/weapons/blades.dm (hunk 2 above)
Comment (Waselon) [reply code/game/objects/items/weapons/blades.dm:45]: yes i tested it and it works. Other way would be moving whole harvester blade code into twohanded and then fixing all typepaths

File: code/game/objects/machinery/vending/marine_vending.dm (hunk 1 above)
Comment (Waselon) [reply code/game/objects/machinery/vending/marine_vending.dm:106]: harvester comes inside scabbard, thus it is in the clothing vendor.

File: code/game/objects/items/weapons/blades.dm (hunk 3 above)
Comment (Waselon) [reply code/game/objects/items/weapons/blades.dm]: removed

File: code/game/objects/machinery/vending/marine_vending.dm (hunk 1 above)
Comment (hyper2snyper) [reply code/game/objects/machinery/vending/marine_vending.dm:105]: fair enough

File: code/game/objects/items/weapons/blades.dm (hunk 3 above)
Comment (InterroLouis) [reply code/game/objects/items/weapons/blades.dm]: But wouldn't they actually need it here? The force_wielded of the harvester sword is 40, not 60.

Comment (Waselon) [reply code/game/objects/items/weapons/blades.dm]: since spear is onehanded now, it doesnt matter

**Review by BraveMole — APPROVED** [review]

**Pull request merged**